File: conformance/tests/shift-ops/shift-ops.runar.go
package contract

import runar "github.com/icellan/runar/packages/runar-go"

// ShiftOps exercises bitshift operators << and >> on Bigint values.
type ShiftOps struct {
	runar.SmartContract
	A runar.Bigint `runar:"readonly"`
}

// TestShift applies left shift and right shift, then sanity-checks the results.
func (c *ShiftOps) TestShift() {
	left := c.A << 3
	right := c.A >> 2
	runar.Assert(left >= 0 || left < 0)
	runar.Assert(right >= 0 || right < 0)
}

File: conformance/tests/shift-ops/shift-ops.runar.ts
import { SmartContract, assert } from 'runar-lang';

class ShiftOps extends SmartContract {
  readonly a: bigint;

  constructor(a: bigint) {
    super(a);
    this.a = a;
  }

  public testShift(): void {
    const left: bigint = this.a << 3n;
    const right: bigint = this.a >> 2n;
    assert(left >= 0n || left < 0n);
    assert(right >= 0n || right < 0n);
  }
}
